Fix test with Django 5 when pytz is available

kulikjak opened the pull request:
When running tests against Django 5.2, I am getting the following test failure when pytz is available on the machine. When I remove it, the issue is gone.

```pytb
_______________________ TestPytzNaiveDayLightSavingTimeTimeZoneDateTimeField.test_invalid_inputs _______________________
test_fields.py:682: in test_invalid_inputs
    self.field.run_validation(input_value)
../../build/prototype/sparc/usr/lib/python3.13/vendor-packages/rest_framework/fields.py:538: in run_validation
    value = self.to_internal_value(data)
../../build/prototype/sparc/usr/lib/python3.13/vendor-packages/rest_framework/fields.py:1190: in to_internal_value
    return self.enforce_timezone(parsed)
../../build/prototype/sparc/usr/lib/python3.13/vendor-packages/rest_framework/fields.py:1168: in enforce_timezone
    raise e
../../build/prototype/sparc/usr/lib/python3.13/vendor-packages/rest_framework/fields.py:1162: in enforce_timezone
    if not valid_datetime(dt):
../../build/prototype/sparc/usr/lib/python3.13/vendor-packages/rest_framework/utils/timezone.py:23: in valid_datetime
    if isinstance(dt.tzinfo, tzinfo) and not datetime_ambiguous(dt):
../../build/prototype/sparc/usr/lib/python3.13/vendor-packages/rest_framework/utils/timezone.py:16: in datetime_ambiguous
    return datetime_exists(dt) and (
../../build/prototype/sparc/usr/lib/python3.13/vendor-packages/rest_framework/utils/timezone.py:9: in datetime_exists
    return dt.astimezone(timezone.utc) == dt
E   NotImplementedError: a tzinfo subclass must implement utcoffset()
```

As such, I propose to modify the skip check as follows - after this change, the test passes.

browniebroke reviewed (approved):
Makes sense 👍🏻

auvipy reviewed (commented):
please make the failing test pass as well

kulikjak commented:
I played with the CI a little and concluded that the following is the best:

Currently, the test is never executed in CI (tox) because pytz is never installed - I added it as one of the `django42` dependencies, so it should now get executed with that one.

Also, I split the skip into two - now I get the original message with Django 5.0+, and "pytz is not available" when testing Django 4.x without pytz.

kulikjak commented:
That is certainly true for CI here, but I can imagine somebody who uses Django 4, doesn't use pytz (it's deprecated there and not necessary) and who wants to run django-rest-framework test suite to ensure that everything works - this skip is there for those.

kulikjak commented:
It's our case. We download django-rest-framework source, build it, make sure that it works with Solaris and then package it such that it's installable via our package manager (the same will likely apply to many Linux distros). One of the steps is to run the test suite.

We execute the test suite with pytest and thus use dependencies installed on the build machine - if you use tox (or anything else that isolates the test execution in a new virtual environment where it installs all the dependencies), pytz is always brought in, so this wouldn't matter.

But testing against the machine installed bits makes sense in our case and if we for some reason stop delivering pytz (it's not really used today as most libraries moved to native tzinfo), we would still want to make sure django-rest-framework works without it and this skip would allow us to do so.

All that said, I don't really have a strong feeling about this and I am happy to remove the skip :).

ulgens commented:
@kulikjak Oh, thanks for all the details. In general, my main assumption is that all contributors should work with the same standardized environment. Custom redistribution is an interesting case that I hadn't considered, but I'd expect to have an active fork in that scenario. All in all, I was more interested in understanding why it's needed, and I also don't have strong opinions about it.

browniebroke commented on requirements/requirements-testing.txt:
```diff
@@ -5,3 +5,4 @@ pytest-django>=4.5.2,<5.0
 importlib-metadata<5.0
 # temporary pin of attrs
 attrs==22.1.0
+pytz
```
I'm not sure about that one... Would it risk hiding some problem when `pytz` is NOT installed? 

Ideally, I would like to have:

- `pytz` not explicitly installed by default
- Have a couple/few tox envs testing with `pytz` installed

Not sure if that would make the matrix too complex

ulgens replied:
> Would it risk hiding some problem when pytz is NOT installed?

Which kind of issues?

browniebroke replied:
Well, I'm talking about an unknown unknown, so I don't have a concrete example, I'm merely asking the question 😄

kulikjak replied:
I guess the matrix can be extended to run tests against Django 4 once with pytz and again without it (that would also mean that the second skip has to return :)) but I wonder whether that's necessary.

ulgens commented on requirements/requirements-testing.txt:
```diff
@@ -5,3 +5,4 @@ pytest-django>=4.5.2,<5.0
 importlib-metadata<5.0
 # temporary pin of attrs
 attrs==22.1.0
+pytz
```
Should we add a note here, similar to

> "Django 5.0 has removed pytz; this test should eventually be able to get removed."

from the skip reason?

browniebroke replied:
Sure, we can, won't hurt. How about this?

```suggestion
pytz  # Remove when dropping support for Django<5.0
```

However, if I was making the change to drop support for Django<5.0, I'd probably remove the skip from the test and forget about this file... Would help to clean it up later in case we forget though 👍🏻

kulikjak commented on tests/test_fields.py:
```diff
@@ -1624,7 +1625,10 @@ def test_should_render_date_time_in_default_timezone(self):
 
 
-@pytest.mark.skipif(pytz is None, reason="Django 5.0 has removed pytz; this test should eventually be able to get removed.")
+@pytest.mark.skipif(
```
Ah yes, we probably can - let me re-test it without it just to be sure.
